client/components/Checkbox.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class Checkbox extends React.PureComponent {
+  render () {
+    const {
+      className,
+      checked,
+      readOnly,
+      label,
+      name,
+      value,
+      onCheck
+    } = this.props;
+
+    const id = `checkbox-${value}-${this.props.id}`;
```

Comment:
If someone passes an id prop, we should use that id unaltered, otherwise things like ID selectors in CSS won't work.

That being said, this is an internal component, and no one should ever ever use ID selectors in CSS, so that's not such a big issue.

This might be easier if you wrapped the `<input>` with the `<label>`, that way we'd get the accessibility and such of the label without having to futz with id stuff.

Reply:
id is taken in as a string and directly used as the element id without manipulation now.